autogen/agentchat/conversable_agent.py

```diff
@@ -2259,14 +2259,19 @@ def generate_init_message(self, message: Union[Dict, str, None], **kwargs) -> Un
         """
         if message is None:
             message = self.get_human_input(">")
+
         if isinstance(message, str):
             return self._process_carryover(message, kwargs)
+
         elif isinstance(message, dict):
-            message = message.copy()
+            message = copy.deepcopy(message)
```

Comment:
maybe we should copy in the process function instead of here. There is a chance that no change is needed and no copy is needed.

Reply:
The process functions accepts the "content" of the message, so only copying the content won't suffice as the original message will reference the newly copied content (I have a test to catch that). My solution was just to short-circuit the function if no carryover message is found, that way no unnecessary copies are made.